pkg/validate/validate.go

```diff
@@ -68,7 +69,29 @@ func validateContainerService(c *api.OpenShiftManagedCluster, externalOnly bool)
 }
 
 func validateUpdateContainerService(cs, oldCs *api.OpenShiftManagedCluster, externalOnly bool) (errs []error) {
-	if !reflect.DeepEqual(cs, oldCs) {
+	newAgents := make(map[string]*api.AgentPoolProfile)
+	for _, newAgent := range cs.Properties.AgentPoolProfiles {
+		newAgents[newAgent.Name] = newAgent
+	}
+
+	var isScaleUpdate bool
+	for i, old := range oldCs.Properties.AgentPoolProfiles {
+		new, ok := newAgents[old.Name]
+		if !ok {
+			continue
+		}
+		if old.Count != new.Count {
+			isScaleUpdate = true
+		}
+		oldCs.Properties.AgentPoolProfiles[i].Count = new.Count
+	}
+
+	if isScaleUpdate && oldCs.Properties.OrchestratorProfile.OrchestratorVersion != cs.Properties.OrchestratorProfile.OrchestratorVersion {
```

Comment:
This is possibly premature.

Reply:
I'm sure someone will try to do it.  :-) Seems good if we want to limit it from the beginning

Reply:
@jim-minter I thought we wanted to start with as little update surface as possible :)

Reply:
yes; at the moment we're not even allowing OrchestratorVersion to change.  I'd rather we start allowing it to change at the point where we have some confidence that it works.

Reply:
> yes; at the moment we're not even allowing OrchestratorVersion to change. I'd rather we start allowing it to change at the point where we have some confidence that it works.

Seems like we need to discuss whether this is something we want to have for PP (don't we want image updates?). FWIW, when I started testing upgrades a couple of weeks ago, I was able to update from one image to another w/o any problem.